tests/console/yast2_lan.pm

```diff
@@ -34,18 +34,73 @@ sub handle_dhcp_popup {
     }
 }
 
+
+sub check_etc_hosts_update {
+    my $looprun = 1;
+    assert_script_run "cat /etc/hosts";
+    until ($looprun == 3) {
+        script_run("yast2 lan; echo yast2-lan-status-\$? > /dev/$serialdev", 0);
+        assert_screen "yast2_lan";
+        send_key 'alt-i';    # edit NIC
+        assert_screen 'yast_ncurses_network_card_setup';
+        send_key 'alt-t';    # set to static ip
+        assert_screen 'yast_ncurses_set_static_ip';
+        send_key 'tab';
+        if ($looprun == 1) {
+            send_key_until_needlematch('NICsetup_ncurses_IP_empty', 'backspace');    # "delete existing IP if any
+            type_string "192.168.122.10";
+        }
+        send_key 'tab';
+        # "delete existing netmask if any
+        if ($looprun == 1) {
+            send_key_until_needlematch('NICsetup_ncurses_mask_empty', 'backspace');
+            type_string "/24";
+        }
+        send_key 'tab';
+        send_key_until_needlematch('NICsetup_ncurses_host_empty', 'backspace');
+        type_string "susetest.test$looprun";
+        assert_screen 'yast_ncurses_static_ip_set';
+        send_key 'alt-n';
+        wait_still_screen;
+        send_key 'alt-o';
+        wait_still_screen;
+        wait_serial("yast2-lan-status-0", 180) || die "'yast2 lan' didn't finish";
+        script_run("egrep \"192.168.122.10\\ssusetest.test$looprun\\ssusetest\" /etc/hosts", 30)
+          && die "Expected entry : \"192.168.122.10    susetest.test$looprun susetest\" was not found in /etc/hosts";
+        assert_script_run "cat /etc/hosts";
+        $looprun++;
+    }
+    # Revert changes
+    script_run("yast2 lan; echo yast2-lan-status-\$? > /dev/$serialdev", 0);
+    assert_screen "yast2_lan";
+    send_key 'alt-i';
+    assert_screen 'yast_ncurses_network_card_setup';
+    send_key 'alt-y';    # set back to DHCP
+    assert_screen 'yast_ncurses_set_dhcp';
+    send_key 'alt-n';
+    wait_still_screen;
+    send_key 'alt-o';
+    wait_still_screen;
```

Comment:
Same as above, it seems to be 100% same code as above, so we can extract it, maybe even parametrized putting the content of the loop.